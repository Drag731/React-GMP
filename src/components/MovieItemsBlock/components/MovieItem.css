.b-item {
  max-width: 300px;
  margin-top: 30px; }
  .b-item__img-wrap {
    margin-bottom: 10px; }
  .b-item__img-pic {
    width: 100%; }
  .b-item__title-wrap {
    display: flex;
    justify-content: space-between;
    align-items: center; }
  .b-item__title-text {
    color: black;
    font-weight: bold; }
  .b-item__title-year {
    border: 1px solid black;
    border-radius: 5px;
    padding: 3px; }
  .b-item__ganre-wrap {
    color: darkgray; }
