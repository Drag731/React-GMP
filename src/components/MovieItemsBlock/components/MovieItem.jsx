import React from 'react';

const MovieItem = ({item, handleGoToDescriptionPage}) => (
    <div className="movie-items__item b-item">
        <div className="b-item__img-wrap">
            <img className="b-item__img-pic" src={item.poster_path} alt=""/>
        </div>
        <div className="b-item__title-wrap">
            <span className="b-item__title-text">{item.title}</span>
            <span className="b-item__title-year">{item.release_date}</span>
        </div>
        <div className="b-item__ganre-wrap">
            <span className="b-item__ganre-text">{item.genres.join(' & ')}</span>
        </div>
    </div>
);

export default MovieItem;