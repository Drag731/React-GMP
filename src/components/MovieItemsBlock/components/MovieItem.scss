.b-item {
    max-width: 300px;
    margin-top: 30px;
    &__img {
        &-wrap {
            margin-bottom: 10px;
        }
        &-pic {
            width: 100%;
        }
    }
    &__title {
        &-wrap {
            display: flex;
            justify-content: space-between;
            align-items: center;
        }
        &-text {
            color: black;
            font-weight: bold;
        }
        &-year {
            border: 1px solid black;
            border-radius: 5px;
            padding: 3px;
        }
    }
    &__ganre {
        &-wrap {
            color: darkgray;
        }
    }
}