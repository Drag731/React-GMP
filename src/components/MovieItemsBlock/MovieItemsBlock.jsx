import React from 'react';

import MovieItem from './components/MovieItem'

const MovieItemsBlock = ({movies}) => (
    <div className="movie-items">
        {movies && movies.map(item => (
            <MovieItem
                item={item}
                key={item.id}
            />
        ))}
    </div>
);

export default MovieItemsBlock;
