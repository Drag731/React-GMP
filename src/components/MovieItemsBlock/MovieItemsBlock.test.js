import React from 'react';
import renderer from 'react-test-renderer';
import {shallow, configure} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import MovieItemsBlock from './MovieItemsBlock.jsx'
import MovieItem from './components/MovieItem.jsx'

configure({ adapter: new Adapter() });

describe('<MovieItemsBlock  />', () => {
    it('Renders MovieItemsBlock component', () => {
        const tree = shallow(
            <MovieItemsBlock />
        );
        expect(tree.find('div.movie-items')).toHaveLength(1);
    });
    it('expect MovieItem component in MovieItemsBlock', () => {
        const movies = [{id:1}, {id:2}];
        const tree = shallow(
            <MovieItemsBlock movies={movies}/>
        );
        expect(tree.find(MovieItem)).toHaveLength(2);
    });
});
