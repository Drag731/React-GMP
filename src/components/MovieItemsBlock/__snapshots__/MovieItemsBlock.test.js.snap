// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MovieItemsBlock  /> Snapshot one 1`] = `
<div
  className="movie-items"
/>
`;

exports[`<MovieItemsBlock  /> Snapshot two 1`] = `
<div
  className="movie-items"
>
  <div
    className="movie-items__item b-item"
  >
    <div
      className="b-item__img-wrap"
    >
      <img
        alt=""
        className="b-item__img-pic"
        src="https://image.tmdb.org/t/p/w500/ldoY4fTZkGISMidNw60GHoNdgP8.jpg"
      />
    </div>
    <div
      className="b-item__title-wrap"
    >
      <a
        className="b-item__title-link"
        href="/filmzilla/description/447365"
        onClick={[Function]}
      >
        <span
          className="b-item__title-text"
        >
          Guardians of the Galaxy Vol. 3
        </span>
      </a>
      <span
        className="b-item__title-year"
      >
        2020-05-01
      </span>
    </div>
    <div
      className="b-item__ganre-wrap"
    >
      <span
        className="b-item__ganre-text"
      >
        Action & Adventure & Science Fiction
      </span>
    </div>
  </div>
</div>
`;
